load-code/pe/copy-pe-sections.yml

```diff
@@ -0,0 +1,32 @@
+rule:
+  meta:
+    name: copy PE sections
+    namespace: load-code/pe
+    author: "@Ana06"
+    scope: function
+    references:
+      - https://0x00sec.org/t/reflective-dll-injection/3080
+      - https://www.ired.team/offensive-security/code-injection-process-injection/reflective-dll-injection
+    examples:
+      - E4C33AC3638EEF68311F8AC0D72483C7:0x401510
+  features:
+    - and:
+      - offset: 0x6 = IMAGE_NT_HEADERS.FileHeader.NumberOfSections
+      - basic block:
+        - or:
+          - and:
+            - description: IMAGE_FIRST_SECTION(nt_header)
+            - offset: 0x14 = IMAGE_NT_HEADERS.FileHeader.SizeOfOptionalHeader
+            - offset: 0x18 = FileHeader.SizeOfOptionalHeader
+          - and:
+            - description: (DWORD)dll_raw + dos_header->e_lfanew + sizeof(IMAGE_NT_HEADERS) + sizeof(IMAGE_SECTION_HEADER) * i
+            - number: 0x28 = sizeof(IMAGE_SECTION_HEADER)
+            - or:
+              - offset/x32: 0xF8 = sizeof(IMAGE_NT_HEADERS32)
+              - offset/x64: 0x108 = sizeof(IMAGE_NT_HEADERS64)
+      - basic block:
+        - and:
+          - offset: 0xC = IMAGE_SECTION_HEADER.VirtualAddress
+          - offset: 0x14 = IMAGE_SECTION_HEADER.PointerToRawData
+          - offset: 0x10 = IMAGE_SECTION_HEADER.SizeOfRawData
+          - mnemonic: call = call CopyMemory, memcpy or similar function
```

Comment:
this seems pretty loose to me. maybe we wait until we have FLIRT that can match memcpy or something?

this would just as easily match `VirtualProtect` that you might expect to see nearby, but would not be related to copying memory.

Reply:
Discussed offline to remove `mnemonic: call = call CopyMemory, memcpy or similar function` and call this rule `enumerate PE sections`